src/translations.h

```diff
@@ -12,6 +12,15 @@
 #include <vector>
 
 #include "optional.h"
+#include "value_ptr.h"
+
+constexpr int INVALID_LANGUAGE_VERSION = 0;
+
+namespace details
```

Comment:
I feel that `details` namespaces should be named with the thing they are the details of, otherwise they're not really providing much namespacing (I realise that they do still provide a documentary purpose).  So I'd suggest `translation_details` here.

Reply:
My intention of using that namespace was to highlight that it is [an implementation internals/details](https://stackoverflow.com/questions/26546265/what-is-the-detail-namespace-commonly-used-for).  Probably should rename it to `detail` to match the name used in `assign.h` and `cata_io.h`.

Reply:
Renamed the namespace to "detail".